addons/io_hubs_addon/components/definitions/rigid_body.py

```diff
@@ -0,0 +1,168 @@
+        min=0.0,
+        soft_max=10.0,
+    )
+
+    gravity: FloatVectorProperty(
+        name="Gravity", description="Object's Gravity",
+        unit="ACCELERATION",
+        subtype="ACCELERATION",
+        default=(0.0, 0.0, -9.8))
+
+    def gather(self, export_settings, object):
+        props = super().gather(export_settings, object)
+        props['collisionMask'] = [value for i, (value, _label, _desc) in enumerate(
+            collision_masks) if self.collisionMask[i]]
+
+        # prefer to store as an array for new components
+        props['angularFactor'] = [
+            self.angularFactor[0],
+            self.angularFactor[2] if export_settings['gltf_yup'] else self.angularFactor[1],
+            self.angularFactor[1] if export_settings['gltf_yup'] else self.angularFactor[2],
+        ]
+        props['gravity'] = [
+            self.gravity[0],
+            self.gravity[2] if export_settings['gltf_yup'] else self.gravity[1],
+            self.gravity[1] if export_settings['gltf_yup'] else self.gravity[2],
+        ]
+        return props
+
+    def draw(self, context, layout, panel):
+        layout.prop(self, "type")
+
+        if (self.disableCollision and self.collisionGroup != "triggers") or (self.collisionGroup == "triggers" and not self.disableCollision):
+            col = layout.column()
+            # col.alert = True
+            col.label(
+                text="When making triggers you likely want 'Is Trigger' checked and collision group set to 'Triggers'",
+                icon='INFO')
+        layout.prop(self, "collisionGroup")
+        layout.label(text="Collision Mask:")
+        col = layout.column(align=True)
+        for i, (_value, label, _desc) in enumerate(collision_masks):
+            col.prop(self, "collisionMask", text=label, index=i, toggle=True)
+        layout.prop(self, "disableCollision")
+
+        layout.prop(self, "mass")
+        layout.prop(self, "linearDamping")
+        layout.prop(self, "angularDamping")
+        layout.prop(self, "linearSleepingThreshold")
+        layout.prop(self, "angularSleepingThreshold")
+        layout.prop(self, "angularFactor")
+        layout.prop(self, "gravity")
+
+    @classmethod
+    def init(cls, obj):
+        obj.hubs_component_list.items.get('physics-shape').isDependency = True
+
+    def migrate(self, migration_type, panel_type, instance_version, host, migration_report, ob=None):
+        migration_occurred = False
+        if instance_version <= (1, 0, 0):
+            migration_occurred = True
+
+            angularFactor = self.angularFactor.copy()
+            angularFactor = Vector((angularFactor.x, angularFactor.z, angularFactor.y))
+            self.angularFactor = angularFactor
+
+            # If the gravity property has not changed, it will take the default value so we don't swizzle
+            # in that case. Is there any way of checking if the property has been modified?
+            if self.gravity != Vector((0.0, 0.0, -9.8)):
```

Comment:
To check if the property has been modified you should be able to do this:
`if self.get('gravity') is not None:`

Reply:
Oh cool, I didn't know that. Thanks.